lib/fastPath.js

```diff
@@ -37,42 +36,34 @@ function pickRange(start, end, step) {
     if (step) {
         fn = filter((entry, idx) => ((idx === start || ((idx - start) % step) === 0) && idx < end));
     }
-    return (arr => (fn && fn(arr)) || (end !== undefined ? arr.slice(start, end): arr.slice(start)));
+    return (arr => (fn && fn(arr)) || arr.slice(start, end));
 }
 
 function mapAndFilter(fn) {
     return function doMapFilter(arr) {
         let result = [];
-        if (typeof arr !== 'object') {
+        if (typeof arr !== 'object' || arr === null) {
             return result;
         }
 
         let current;
-        Object.keys(arr).forEach(i => ((current = fn(arr[i], i))!== undefined) ? (result[result.length] = current) : undefined);
+        Object.keys(arr).forEach(i => ((current = fn(arr[i], i)) !== undefined) && result.push(current));
         return result;
```

Comment:
this was done the old way specifically for better performance. we benchmarked it to work faster.